rfc6962-bis.xml

```diff
@@ -1396,6 +1402,94 @@ the STH changes.
+          <t>
+            When a client has an STH <spanx style="verb">first_hash</spanx> for tree size <spanx style="verb">first</spanx>, an STH <spanx style="verb">second_hash</spanx> for tree size <spanx style="verb">second</spanx> where <spanx style="verb">0 &lt; first &lt; second</spanx>, and has received a <spanx style="verb">consistency</spanx> array that they wish to use to verify both hashes, the following algorithm may be used:
+            <list style="numbers">
+              <t>If <spanx style="verb">first</spanx> is an exact power of 2, then prepend <spanx style="verb">first_hash</spanx> to the <spanx style="verb">consistency</spanx> array.</t>
+              <t>Set <spanx style="verb">fn</spanx> to <spanx style="verb">first - 1</spanx> and <spanx style="verb">sn</spanx> to <spanx style="verb">second - 1</spanx>.</t>
+              <t>If <spanx style="verb">LSB(fn)</spanx> is set, then right-shift both <spanx style="verb">fn</spanx> and <spanx style="verb">sn</spanx> equally until <spanx style="verb">LSB(fn)</spanx> is not set.</t>
+              <t>Set both <spanx style="verb">fr</spanx> and <spanx style="verb">sr</spanx> to the first value in the <spanx style="verb">consistency</spanx> array.</t>
+              <t>
+                For each subsequent value <spanx style="verb">c</spanx> in the <spanx style="verb">consistency</spanx> array:
+                <vspace blankLines="1" />
+                If <spanx style="verb">LSB(fn)</spanx> is set, or if <spanx style="verb">fn</spanx> is equal to <spanx style="verb">sn</spanx>, then:
+                <list style="numbers">
+                  <t>
+                    Set <spanx style="verb">fr</spanx> to <spanx style="verb">HASH(0x01 || c || fr)</spanx>
+                    <vspace blankLines="0" />
+                    Set <spanx style="verb">sr</spanx> to <spanx style="verb">HASH(0x01 || c || sr)</spanx>
+                  </t>
+                  <t>If <spanx style="verb">LSB(fn)</spanx> is not set, then right-shift both <spanx style="verb">fn</spanx> and <spanx style="verb">fn</spanx> equally until <spanx style="verb">LSB(fn)</spanx> is set.</t>
+                </list>
+                Otherwise:
+                <list style="empty">
+                  <t>Set <spanx style="verb">sr</spanx> to <spanx style="verb">HASH(0x01 || sr || c)</spanx></t>
+                </list>
+                <vspace blankLines="1" />
+                Finally, right-shift both <spanx style="verb">fn</spanx> and <spanx style="verb">sn</spanx> one time.
+              </t>
+              <t>After completing iterating through the <spanx style="verb">consistency</spanx> array as described above, verify that the <spanx style="verb">fr</spanx> calculated is equal to the <spanx style="verb">first_hash</spanx> supplied and that the <spanx style="verb">sr</spanx> calculated is equal to the <spanx style="verb">second_hash</spanx> supplied.</t>
```

Comment:
b) is guaranteed because we're iterating through the consistency array (this differs from our open-source code which does it differently).

a) "fn" and "sn" are means to an end. Does it matter if they are zero or not so long as the hashes calculated both match?  Since we iterate through all elements of the consistency array and always apply to at least "sr" I would expect that comparing "sr" against the supplied hash to be sufficient to verify correct consistency date is supplied.